Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Gender laws overhaul

## Changelog:
- Only females can be theocratic rulers in **Elunic** religion.

## Developer changelog:
- The capital of Nordrassil province is a castle.
- Developed Nordrassil province.
- Females control Elunic temples in Hyjal.

## Script changelog (ignore if you can't read CK2 script):
- Reworked gender-related triggers and moved them to `wc_gender_scripted_triggers`. Now they should make more sense and be less CPU heavy: `is_gender_equal_trigger` no longer checks `is_matriarchal_trigger`.
- Added `religion_allows_to_hold_temples_trigger` in `theocracy_governments.txt`.
**_Note:_** These lines previously caused a bug where female rulers could become nomads. I hope it's not possible now, but still, be careful.
- Added the gender triggers back to `HF.23034`, `HF.23700` and `HF.23701` together with reworking `if`s to make them less CPU heavy.
**_Note:_** It may be CPU heavy, be careful.
- Commented lines for evil characters in `539` event since they're forced to change government in `WCGOV.5000`.



# How to test:
Try to give temples to Elunic females and males, observe the game for +10 years.
Try to notice weird stuff like:
- Elunic males being theocrats.
- Female nomads with one landed barony holding (happened when a female got a holding though her religion didn't allow it, can't reproduce it after the fix, but still).
- The game running too slow (just tell if you feel it became slower).
